### DISCUSSION TITLE
Display information about deprecated versions and packages

### DISCUSSION THREAD

**Discussion by TravisCardwell** [flora-pm/flora-server#318] [Feature Requests]
Package pages do not display any information about deprecated versions or deprecated packages.  Should this information be added?  Personally, I have needlessly spent a significant amount of time in "dependency hell" when I did not notice such deprecation information, I wish it was more prominently visible on Hackage.

As an example of deprecated versions, consider the `mtl` package ([Hackage](https://hackage.haskell.org/package/mtl), [Flora](https://flora.pm/packages/@haskell/mtl)).  Hackage displays deprecated versions using a red font.  When displaying a most recent release that is deprecated, however, the version is displayed in bold black and there is no way to determine that it is a deprecated version without navigating to an older version.  This information is included in the Hackage index tarball; check out the `preferred-versions` files for applicable packages.

As an example of a deprecated package, consider the `ansi-wl-pprint` package ([Hackage](https://hackage.haskell.org/package/ansi-wl-pprint), [Flora](https://flora.pm/packages/@hackage/ansi-wl-pprint)).  Hackage displays package deprecation information at the top of the page, just above the package description.  It is pretty easy to miss, IMHO.  This information is unfortunately not included in the Hackage index tarball, as far as I can tell.  The only way I know to get this information is to use the [Hackage API](https://hackage.haskell.org/api#versions):

```
$ curl \
    -H 'Accept: application/json' \
    https://hackage.haskell.org/packages/deprecated \
  > deprecated.json
$ du -h deprecated.json
76K     deprecated.json
$ jq length deprecated.json
1168
$ jq '.[0]' deprecated.json
```

```json
{
  "deprecated-package": "2captcha",
  "in-favour-of": [
    "captcha-2captcha"
  ]
}
```

**Comment by tchoutri** [comment]
Indeed it looks quite easy to retrieve this data. 
I'll open a low-priority ticket (for the moment), as it looks like a good starting point for someone to get acquainted with the codebase.

**Comment by tchoutri** [comment]
Here is the ticket: https://github.com/flora-pm/flora-server/issues/319

**Comment by tchoutri** [comment]
@TravisCardwell Would you be interested in working on this?

**Comment by TravisCardwell** [comment]
Though interested, I currently lack time to work on it.

I took a look at the issue, and it looks nice!  Great project management!

Regarding pruning deprecated packages from search listings, I think that it is more developer-friendly to display deprecated packages but clearly mark them as deprecated.  For example, the `ansi-wl-pprint` package is deprecated yet still widely used because it is still used by `optparse-applicative`.  A developer that is using the package may be confused if they search for it and are unable to find it.  (This is the current situation on Hackage.)

**Reply by tchoutri** [reply]
I gotchu :)

**Comment by tchoutri** [comment]
closed by #319
